src/docker_snapshot/__init__.py

```diff
@@ -1,10 +1,15 @@
+import typing as t
 import click
 from click_aliases import ClickAliasedGroup
 from rich.console import Console
 from rich.table import Table
 from docker_snapshot import container, settings, snapshot, utils
 
 
+if t.TYPE_CHECKING:
+    from typing_extensions import TypeGuard
```

Comment:
Note: this is only required due to Python 3.9 support. It's available under `typing` as of 3.10 (IIRC).

Reply:
Okay, was going to ask, makes sense

Reply:
Soo, the point of the `if t.TYPE_CHECKING` is that it is only true when the type checker is running, right? And here we import `TypeGuard` that is then used inside a string to circumvent import failure during normal runtime?

Reply:
> the point of the `if t.TYPE_CHECKING` is that it is only true when the type checker is running, right?

Correct.

> And here we import `TypeGuard` that is then used inside a string to circumvent import failure during normal runtime?

Also correct, and the "inside a string" can be a bit confusing. With proper tooling you can still inspect the type even if it's wrapped inside a string. The other alternative is `from __future__ import annotations` (see [PEP 563](https://peps.python.org/pep-0563)), but this is my personal preference.

Reply:
Maybe we should just drop 3.9? Then we could drop the whole "inside a string" thing, right?

Reply:
I pushed a change dropping 3.9 support, what do you think?

Reply:
> Then we could drop the whole "inside a string" thing, right?

This can also be avoided by using `from __future__ import annotations`, but up to you, I'm fine with `>=3.10`.